### PR TITLE
Improved speed of posting entries and comments

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
Speedup the entry & comment creation by generating notifications in subtask

also 

Added failfast and improved linter checks

- Fixes Z#1422

### CODE REVIEW

File: src/django/VLE/tasks/notifications.py [hunk 1]
```diff
@@ -23,3 +23,37 @@ def generate_new_node_notifications(node_ids):
                     user=author.user,
                     node=node,
                 )
+
+
+@shared_task
+def generate_new_comment_notifications(comment_id):
+    comment = VLE.models.Comment.objects.get(pk=comment_id)
```
Comment (mjvkeulen) [inline src/django/VLE/tasks/notifications.py:30]: Prob want to use select related on the author and journal and stuff to save some queries.

File: src/django/VLE/tasks/notifications.py [hunk 2]
```diff
@@ -23,3 +23,37 @@ def generate_new_node_notifications(node_ids):
...
+            comment=comment,
+        )
+
+    # Generate notifications for all other students in journal
+    for author in comment.entry.node.journal.authors.all().exclude(user=comment.author):
+        VLE.models.Notification.objects.create(
+            type=VLE.models.Notification.NEW_COMMENT,
+            user=author.user,
+            comment=comment,
+        )
```
Comment (mjvkeulen) [inline src/django/VLE/tasks/notifications.py:46]: Bulk create? Or was the create function overloaded?

Comment (Larspolo) [reply src/django/VLE/tasks/notifications.py:46]: heavily overloaded

File: src/django/VLE/tasks/notifications.py (hunk 1 above)
Comment (Larspolo) [reply src/django/VLE/tasks/notifications.py:30]: Its in a shared task, so it doesnt matter too muhc, but its a good habbit to have, so Ill update it

Comment (Larspolo) [reply src/django/VLE/tasks/notifications.py:30]: Nvm, only queries have select related, and other comment values get used in the notification creation

Comment (mjvkeulen) [reply src/django/VLE/tasks/notifications.py:30]: This is a query right?

Comment (Larspolo) [reply src/django/VLE/tasks/notifications.py:30]: `.get` returns the comment object, not a *queryset

Comment (mjvkeulen) [reply src/django/VLE/tasks/notifications.py:30]: But it's still a query and therefore possible. Just flip the order, Comment.objects.select_related('etc'').get(pk=comment_id)

Comment (Larspolo) [reply src/django/VLE/tasks/notifications.py:30]: Didnt know that, added it now

**Review by engelhamer — APPROVED** [review]

File: src/django/test/test_notification.py [hunk 3]
```diff
@@ -157,14 +160,30 @@ def test_entry_notification(self):
...
+        node.save()
+        entry.refresh_from_db()
+        assert not hasattr(entry, 'node'), 'test setup should remove node from entry'
+        with QueryContext() as context:
+            generate_new_entry_notifications(entry.pk, node.pk)
+        node.entry = entry
+        node.save()
+        entry.refresh_from_db()
+        assert hasattr(entry, 'node'), 'test setup should add node to entry'
+        with self.assertNumQueries(len(context.captured_queries) - 2):
```
Comment (mjvkeulen) [inline src/django/test/test_notification.py:184]: What does the -2 signify?

Comment (Larspolo) [reply src/django/test/test_notification.py]: Put it in variable

File: src/django/test/test_notification.py [hunk 4]
```diff
@@ -157,14 +160,31 @@ def test_entry_notification(self):
...
 
         self.check_send_push_notification(new_notifactions.get(type=Notification.NEW_ENTRY))
 
+        # Because of a limitation in the test factory, we also need to provide the node pk
+        # However, when entry already has a node, it should not be used to reduce the number of queries used
+        node = entry.node
+        node.entry = None
+        node.save()
+        entry.refresh_from_db()
+        assert not hasattr(entry, 'node'), 'test setup should remove node from entry'
```
Comment (mjvkeulen) [inline src/django/test/test_notification.py:177]: Isn't this basically testing Django internals? (it increases the strain on reading the test to me)

File: src/django/test/test_notification.py [hunk 5]
```diff
@@ -157,14 +160,31 @@ def test_entry_notification(self):
...
+        with QueryContext() as context:
+            generate_new_entry_notifications(entry.pk, node.pk)
+        node.entry = entry
```
Comment (mjvkeulen) [inline src/django/test/test_notification.py:180]: Hspace as this is a new test block? Could even move your comment here -> # However, when entry already has a node, it should not be used to reduce the number of queries used

File: src/django/test/test_notification.py [hunk 6]
```diff
@@ -157,14 +160,31 @@ def test_entry_notification(self):
...
+        node.save()
+        entry.refresh_from_db()
+        assert hasattr(entry, 'node'), 'test setup should add node to entry'
+        nr_queries_saved_select_related_and_node_fetching = 2
```
Comment (mjvkeulen) [inline src/django/test/test_notification.py:184]: Select related does not increase the query count.

Comment (Larspolo) [reply src/django/test/test_notification.py:184]: is "saves" queries

**Review by mjvkeulen — APPROVED** [review]

**Pull request merged**